index.js

```diff
@@ -0,0 +1,224 @@
+'use strict'
+
+const fp = require('fastify-plugin')
+const yaml = require('js-yaml')
+const path = require('path')
+
+function fastifySwagger (fastify, opts, next) {
+  fastify.decorate('swagger', swagger)
+
+  opts = opts || {}
+  opts.swagger = opts.swagger || {}
+
+  const info = opts.swagger.info || null
+  const host = opts.swagger.host || null
+  const schemes = opts.swagger.schemes || null
+  const consumes = opts.swagger.consumes || null
+  const produces = opts.swagger.produces || null
+  const basePath = opts.swagger.basePath || null
+
+  const exposeRoute = !(opts.exposeRoute === false)
+  if (exposeRoute) {
+    fastify.get(opts.route || '/documentation', {
+      schema: {
+        description: 'Documentation route, if yaml is true the payload will be in yaml format',
+        querystring: {
+          type: 'object',
+          properties: {
+            yaml: { type: 'boolean' }
+          }
+        },
+        response: {
+          200: {
+            description: 'The swagger definition of the routes',
+            type: 'object',
+            properties: {
+              hello: { type: 'string' }
+            }
+          }
+        }
+      }
+    }, function (req, reply) {
+      if (req.query.yaml) {
+        return reply
+          .header('Content-Type', 'text/plain')
+          .send(fastify.swagger({ yaml: true }))
+      }
+      reply.send(fastify.swagger())
+    })
+  }
+
+  function swagger (opts) {
+    const swaggerObject = {}
+
+    // Base swagger info
+    // this info is displayed in the swagger file
+    // in the same order as here
+    swaggerObject.swagger = '2.0'
+    if (info) {
+      swaggerObject.info = info
+    } else {
+      swaggerObject.info = {
+        version: '1.0.0',
+        title: require(path.join(__dirname, 'package.json')).name || ''
+      }
+    }
+    if (host) swaggerObject.host = host
+    if (schemes) swaggerObject.schemes = schemes
+    if (basePath) swaggerObject.basePath = basePath
+    if (consumes) swaggerObject.consumes = consumes
+    if (produces) swaggerObject.produces = produces
+
+    swaggerObject.paths = {}
+
+    for (var node of fastify) {
+      // The node path name
+      const url = formatParamUrl(Object.keys(node)[0])
+      // object with all the methods of the node
+      const routes = node[Object.keys(node)[0]]
+      const swaggerRoute = {}
+      swaggerRoute[url] = {}
+
+      // let's iterate over the methods
+      Object.keys(routes).forEach(method => {
```

Comment:
I think you might have way better perf using a `for`  loop. You are allocating a new function for each node in fastify.